src/FSharpVSPowerTools.Core/UntypedAstUtils.fs

```diff
@@ -808,6 +807,32 @@ module Outlining =
             | Simple _ -> yield! Seq.collect visitSynMemberDefn members
         }
 
+    let private collectOpens (decls: SynModuleDecls) =
+        let mergeRanges (ranges: range list) = 
+            match ranges with
+            | last :: _ :: _ ->
+                let first = List.rev ranges |> List.head
+                Some (Range.mkRange "" first.Start last.End)
+            | _ -> None
+
+        decls
+        |> List.choose (function SynModuleDecl.Open (_, r) -> Some r | _ -> None)
+        |> List.fold (fun (res, currentRanges) r ->
+            match currentRanges with
+            | [] -> res, [r]
+            | lastRange :: _ when r.StartLine = lastRange.EndLine + 1 -> res, r :: currentRanges
```

Comment:
I think the assumption `r.StartLine = lastRange.EndLine + 1` isn't always correct.

I think it maybe simpler to write a custom recursive function that:

- collect all consecutive occurrences of `open` into sublists.
- merge ranges of each sublists.